File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    const data1 = 'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz'
    const data2 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ'

    let data = s.split('')
    let answer = ""
    for(const item of data){
        if(item === ' ') answer += " "
        
        else if(data1.indexOf(item)!== -1){
            answer += data1[data1.indexOf(item) + n]
        }
        else{
            answer += data2[data2.indexOf(item) + n]
        }
    }
    return answer 
}


// 1. 일단 문자열을 배열로 쪼개서 각 루프를 돌아야함 .
// 2. 루프에서 n만큼 점프를 시켜야하는데 기준점을 알고 

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
import java.util.*;

class Solution {
    public String solution(String s, int n) {
        String upperCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String lowerCase = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";
        
        String answer = "";
        
        for(int i =0; i<s.length(); i++){
            char nowLetter = s.charAt(i);
            int index;
            if(nowLetter == ' ') answer += ' ';
            //1. 대소문자 구분후 인덱스 찾기
            else if(Character.isUpperCase(nowLetter)){
                index = upperCase.indexOf(nowLetter);
                
                answer += String.valueOf(upperCase.charAt(index+n));
            }
            else{
                index = lowerCase.indexOf(nowLetter);
                answer += String.valueOf(lowerCase.charAt(index+n));
            }
        
        }
                                         
        return answer;
    }
    
}
